ob1k-cache/src/test/java/com/outbrain/ob1k/cache/memcache/folsom/CompressionTranscoderTest.java

```diff
@@ -0,0 +1,38 @@
+package com.outbrain.ob1k.cache.memcache.folsom;
+
+import com.outbrain.ob1k.cache.memcache.compression.CompressionAlgorithm;
+import com.spotify.folsom.Transcoder;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.runners.MockitoJUnitRunner;
+
+/**
+ * Created by bshushi on 26/03/2017.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class CompressionTranscoderTest {
```

Comment:
the tests we really want here is to check all compressions types as you did, using directly the compression implementation, without doing any mocking - just providing sample byte[], checking compress & de-compress works. even better - define a sample value that should be compressed, and check the size.

about the CompressionTranscoder test, it should be really thin test just checking the contract of the class - provide mocked delegator and mocked compressor and just make sure the contract is being valid.